src/App.jsx

```diff
@@ -14,70 +17,79 @@ export const goods = [
-    <h1 className="title is-flex is-align-items-center">No goods selected</h1>
-
-    <h1 className="title is-flex is-align-items-center">
-      Jam is selected
-
-      <button
-        data-cy="ClearButton"
-        type="button"
-        className="delete ml-3"
-      />
-    </h1>
-
-    <table className="table">
-      <tbody>
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
-
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Dumplings
-          </td>
-        </tr>
+export const App = () => {
+  const [selectedGood, changeGood] = useState('Jam');
 
-        <tr data-cy="Good" className="has-background-success-light">
-          <td>
-            <button
-              data-cy="RemoveButton"
-              type="button"
-              className="button is-info"
-            >
-              -
-            </button>
-          </td>
+  return (
+    <main className="section container">
+      <h1 className="title is-flex is-align-items-center">
+        {
+          selectedGood
+            ? (
+              <>
+                {`${selectedGood} is selected`}
 
-          <td data-cy="GoodTitle" className="is-vcentered">
-            Jam
-          </td>
-        </tr>
+                <button
+                  onClick={() => changeGood('')}
+                  data-cy="ClearButton"
+                  type="button"
+                  className="delete ml-3"
+                />
+              </>
+            )
+            : ('No goods selected')
+        }
+      </h1>
 
-        <tr data-cy="Good">
-          <td>
-            <button
-              data-cy="AddButton"
-              type="button"
-              className="button"
-            >
-              +
-            </button>
-          </td>
+      <table className="table">
+        <tbody>
+          {
+            goods.map(good => (
+              <tr
+                data-cy="Good"
+                className={cn({
+                  'has-background-success-light': selectedGood === good,
+                })}
+                key={good}
+              >
+                <td>
+                  {
+                    selectedGood === good
+                      ? (
+                        <button
+                          onClick={() => {
+                            changeGood('');
+                          }}
+                          data-cy="RemoveButton"
+                          type="button"
+                          className="button is-info"
+                        >
+                          -
+                        </button>
+                      )
+                      : (
+                        <button
+                          onClick={() => {
+                            changeGood(good);
+                          }}
```

Comment:
```suggestion
                          onClick={() => selectGood(good)}
```
Create a separate function to handle this event. Though you'd do the same in a separate function if you move this inside a function your code will be easier to maintain and extend.